datatable/README.md

```diff
@@ -155,7 +155,20 @@ domain. Doing this has the following benefits:
 2. Document and evolve your ubiquitous domain language
 3. Enforce certain patterns
 
-A custom table type can be registered as follows:
+There are two helpers for defining custom table types:
+
+```java
+// Defines a DataTableType that converts an entry (map of header name to row value) 
+// to an object, using reflection.
+registry.defineDataTableType(DataTableType#entry(Class))
+
+// Defines a DataTableType that converts a single cell
+// to an object, by calling its `String` constructor (if it exists).
+registry.defineDataTableType(DataTableType#cell(Class))
```

Comment:
Can we add an implementation note here: Something along the lines of we're using Jackson, but don't rely on anything but getters/setter/constructor/fromString/valueOf methods?